src/main/java/de/tum/in/www1/artemis/service/ComplaintResponseService.java

```diff
@@ -43,51 +42,212 @@ public ComplaintResponseService(ComplaintRepository complaintRepository, Complai
+            throw new IllegalArgumentException("Complaint id should not be null");
         }
-        if (complaintResponse.getComplaint() == null || complaintResponse.getComplaint().getId() == null) {
-            throw new BadRequestAlertException("A complaint response can be only associated to a complaint", ENTITY_NAME, "noresultid");
+        Optional<Complaint> complaintFromDatabaseOptional = complaintRepository.findByIdWithEagerAssessor(complaintId);
+        if (complaintFromDatabaseOptional.isEmpty()) {
+            throw new IllegalArgumentException("Complaint was not found in database");
         }
+        Complaint complaintFromDatabase = complaintFromDatabaseOptional.get();
+        ComplaintResponse unfinishedComplaintResponse = getUnfinishedComplaintResponse(complaintFromDatabase);
 
-        Long complaintId = complaintResponse.getComplaint().getId();
+        Result originalResult = complaintFromDatabase.getResult();
+        User assessor = originalResult.getAssessor();
+        User user = this.userService.getUser();
+        StudentParticipation studentParticipation = (StudentParticipation) originalResult.getParticipation();
+        if (!isUserAuthorizedToRespondToComplaint(studentParticipation, complaintFromDatabase, assessor, user)) {
+            throw new AccessForbiddenException("Insufficient permission for removing the lock on the complaint");
+        }
+        // only instructors and the original reviewer can remove the lock while it is still running
+        if (blockedByLock(unfinishedComplaintResponse, user, studentParticipation)) {
+            throw new ComplaintResponseLockedException(unfinishedComplaintResponse);
+        }
+        complaintResponseRepository.deleteById(unfinishedComplaintResponse.getId());
+        log.debug("Removed empty complaint and thus lock for complaint with id : {}", complaintId);
+    }
 
-        // Do not trust user input
-        Optional<Complaint> originalComplaintOptional = complaintRepository.findByIdWithEagerAssessor(complaintId);
-        if (originalComplaintOptional.isEmpty()) {
-            throw new BadRequestAlertException("The complaint you are referring to does not exist", ENTITY_NAME, "noresult");
+    private ComplaintResponse getUnfinishedComplaintResponse(Complaint complaintFromDatabase) {
+        if (complaintFromDatabase.isAccepted() != null) {
+            throw new IllegalArgumentException("Complaint is already handled");
         }
+        if (complaintFromDatabase.getComplaintResponse() == null) {
+            throw new IllegalArgumentException("Complaint response does not exists for given complaint");
+        }
+        if (complaintFromDatabase.getComplaintResponse().getSubmittedTime() != null) {
+            throw new IllegalArgumentException("Complaint response is already submitted");
+        }
+        ComplaintResponse unfinishedComplaintResponse = complaintFromDatabase.getComplaintResponse();
+        return unfinishedComplaintResponse;
```

Comment:
```suggestion
        return complaintFromDatabase.getComplaintResponse();
```

Reply:
Done in 72fed0ffad1de5e7dc5f64debe96b4ba8e016aec